x/pairing/keeper/msg_server_relay_payment.go

```diff
@@ -231,6 +233,32 @@ func (k msgServer) RelayPayment(goCtx context.Context, msg *types.MsgRelayPaymen
 			details["rewardedCU"] = strconv.FormatInt(reward.Quo(coinsPerCu).TruncateInt64(), 10)
 		}
 
+		// track the provider's CU (after QoS influence)
+		proj, err := k.projectsKeeper.GetProjectForDeveloper(ctx, clientAddr.String(), uint64(relay.Epoch))
+		if err != nil {
+			return nil, utils.LavaFormatError("critical: cannot get client project", legacyerrors.ErrKeyNotFound,
+				utils.Attribute{Key: "client", Value: clientAddr.String()},
+				utils.Attribute{Key: "block", Value: strconv.FormatUint(uint64(relay.Epoch), 10)},
+			)
+		}
+		sub, found := k.subscriptionKeeper.GetSubscription(ctx, proj.Subscription)
+		if !found {
+			return nil, utils.LavaFormatError("critical: cannot get client subscription", legacyerrors.ErrKeyNotFound,
+				utils.Attribute{Key: "sub_consumer", Value: proj.Subscription},
+			)
+		}
+		cuAfterQos := uint64(reward.Quo(coinsPerCu).TruncateInt64())
+		err = k.subscriptionKeeper.AddTrackedCu(ctx, sub.Consumer, relay.Provider, relay.SpecId, cuAfterQos, sub.Block)
+		if err != nil {
+			return nil, err
+		}
+
+		utils.LogLavaEvent(ctx, logger, subscriptiontypes.AddTrackedCuEventName, map[string]string{
```

Comment:
lets use the already exists event here, it might already contain the relevant fields

Reply:
done. see https://github.com/lavanet/lava/commit/ddf3113fc8632e5dea14e5825f1f26c914758a13